include/game/entity.h

```diff
@@ -67,8 +50,167 @@ u32 sub_800DF38(Sprite *, s32, s32, Player *);
 u32 sub_800C944(Sprite *, s32, s32);
 
 // TODO: Include header this belongs to
+bool32 sub_800CA20(Sprite *s, s32 x, s32 y, u16 p3, Player *p);
 u32 sub_800CCB8(Sprite *, s32 x, s32 y, Player *);
 
 void sub_801FD34(s32, s32, s32);
 
+// After a MapEntity is initialized, its x-value in the layout-data gets set to -2.
+#define MAP_ENTITY_STATE_INITIALIZED (-2)
+#define MAP_ENTITY_STATE_MINUS_THREE (-3)
+#define SET_MAP_ENTITY_INITIALIZED(mapEnt)                                              \
+    {                                                                                   \
+        s32 negativeTwo;                                                                \
+        s16 forMatching;                                                                \
+        negativeTwo = MAP_ENTITY_STATE_INITIALIZED;                                     \
+        forMatching = negativeTwo;                                                      \
+        mapEnt->x = forMatching;                                                        \
+    }
+
+#define SET_MAP_ENTITY_NOT_INITIALIZED(mapEnt, initialX)                                \
+    {                                                                                   \
+        mapEnt->x = initialX;                                                           \
+    }
+
+#define ENTITY_INIT(eType, eName, _task, _taskPrio, _taskFlags, _taskDtor)              \
+    struct Task *t                                                                      \
+        = TaskCreate(_task, sizeof(eType), _taskPrio, _taskFlags, _taskDtor);           \
+    eType *eName = TaskGetStructPtr(t);                                                 \
+    Sprite *s = &eName->s;                                                              \
+    eName->base.regionX = spriteRegionX;                                                \
+    eName->base.regionY = spriteRegionY;                                                \
+    eName->base.me = me;                                                                \
+    eName->base.spriteX = me->x;                                                        \
+    eName->base.spriteY = spriteY;
+
+#define ENTITY_INIT_2(eType, eName, _task, _taskPrio, _taskFlags, _taskDtor,            \
```

Comment:
Commented before, but not a fan of these init macros. Yes we can a small amount of code reuse, but for what? It's not complex and the macro makes things more confusing.

Sometimes copy and pasted code is easier to understand and modify than a macro with so many inputs and different uses for different attributes.

The rest of the macros in here are good! :)

Reply:
Will think about it.
It definitely is less explicit.